src/main/scala/com/wavesplatform/history/HistoryWriterImpl.scala

```diff
@@ -40,29 +42,35 @@ class HistoryWriterImpl private(file: Option[File], val synchronizationToken: Re
 
   private def displayFeatures(s: Set[Short]): String = s"FEATURE${if (s.size > 1) "S"} ${s.mkString(", ")} ${if (s.size > 1) "WERE" else "WAS"}"
 
+  private def featureVotesCountInActivationWindow(windowClosingHeight: Int, ngState: Option[NgState] = None): Map[Short, Int] = read { implicit lock =>
```

Comment:
1. Please don't use default values
2. `featureVotesCountInActivationWindow` as well as `status` should not know anything about `NgState`. Please refactor that.